container/container_unix.go

```diff
@@ -550,6 +551,75 @@ func (container *Container) IpcMounts() []execdriver.Mount {
 	return mounts
 }
 
+func updateCommand(c *execdriver.Command, resources runconfig.Resources) {
+	c.Resources.BlkioWeight = resources.BlkioWeight
+	c.Resources.CPUShares = resources.CPUShares
+	c.Resources.CPUPeriod = resources.CPUPeriod
+	c.Resources.CPUQuota = resources.CPUQuota
+	c.Resources.CpusetCpus = resources.CpusetCpus
+	c.Resources.CpusetMems = resources.CpusetMems
+	c.Resources.Memory = resources.Memory
+	c.Resources.MemorySwap = resources.MemorySwap
+	c.Resources.MemoryReservation = resources.MemoryReservation
+	c.Resources.KernelMemory = resources.KernelMemory
+}
+
+// SetContainer sets resources of a container.
+func (container *Container) SetContainer(hostConfig *runconfig.HostConfig) error {
+	container.Lock()
+
+	resources := hostConfig.Resources
+	cResources := &container.HostConfig.Resources
+	if resources.BlkioWeight != 0 {
```

Comment:
I might be misunderstanding but I don't think this PR can be merged w/o the use of pointers. Take as an example "resources.MemorySwap", I think that will be zero if not set on the "docker set" command line, but zero is a valid value according to the docs. Wouldn't this need a pointer to know if it was included in the json?